### DISCUSSION TITLE
Numeric types prefixes/suffixes and scientific notation

### DISCUSSION THREAD

**Discussion by ghost** [objeck/objeck-lang#111] [General]
The Getting Started page has no information regarding this:

https://objeck.org/getting_started.html#lits_vars

C has a very detailed reference:

https://en.cppreference.com/w/cpp/language/integer_literal

https://en.cppreference.com/w/cpp/language/floating_literal

**Comment by objeck** [comment]
Thanks, Objeck follows the C standard. I will update the guide.

**Reply by 1997tn** [reply]
> Objeck follows the C standard

Is it already the case or just raised after this discussion? Objeck doesn't like C. Please don't learn C's bad habits. Objeck only has Int (4 byte integer equivalent to C's int) and Float (8 byte floating point equivalent to C's double) so there is no needs to distinguish between different types of integers or floating point numbers. Unless you are going to introduce stuffs like LongInt (8 byte integer) or LongFloat or BigInt (Java's BigInteger) or arbitrary precision floating point number BigFloat (Java's BigDecimal). Even if so, the type prefixes and suffixes will be definitely different from C. Please consider @objeck, don't learn C's bad habits. With the current supported types on the table, there is no need for type prefixes/suffixes on Objeck.

**Reply by ghost** [reply]
Let me clarify. I don't ask for C-like types prefixes/suffixes. I'm going to help with this issue https://github.com/lite-xl/lite-xl/issues/1352 (creating an Objeck lexer for Lite XL). I only wanted to know if there is any types prefixes/suffixes currently _in use_. If there is none then it's much better for me, as I don't have to write the patterns to match various types prefixes/suffixes like when dealing with a C-like language.

**Reply by ghost** [reply]
What's the syntax for binary number, octal number and hex number? Is it the same as C? 0b prefix for binary number, 0 prefix for octal number and 0x prefix for hex number?

**Comment by objeck** [comment]
@iqskpduswupkcjqg, Thanks for the assistance. 

Please refer to the [strtol](https://cplusplus.com/reference/cstdlib/strtol/) C function. Objeck [uses](https://github.com/objeck/objeck-lang/blob/master/core/compiler/scanner.cpp#L677) the wide-character version of this function to convert strings to integers. 

I should clarify that Objeck reads and writes integer literals as 32-bit values however internally, integers and doubles are both 64-bits on all targets.

**Comment by objeck** [comment]
I added [binary literal support](https://github.com/objeck/objeck-lang/commit/45d18329be2c77bb5cdf6799c0f6ad1d36a47abc) per the C++14 standard
